frontend/src/integrations/stackoverflow/components/stackoverflow-connect-drawer.vue

```diff
@@ -0,0 +1,400 @@
+      <div
+        class="flex grow items-center"
+        :class="
+          hasFormChanged ? 'justify-between' : 'justify-end'
+        "
+      >
+        <el-button
+          v-if="hasFormChanged"
+          class="btn btn-link btn-link--primary"
+          @click="doReset"
+          ><i class="ri-arrow-go-back-line"></i>
+          <span>Reset changes</span></el-button
+        >
+        <div class="flex gap-4">
+          <el-button
+            class="btn btn--md btn--bordered"
+            @click="doCancel"
+          >
+            <app-i18n code="common.cancel"></app-i18n>
+          </el-button>
+          <el-button
+            :loading="isVolumeUpdating"
+            class="btn btn--md btn--primary"
+            :class="{
+              disabled: !hasFormChanged || connectDisabled
+            }"
+            @click="hasFormChanged ? connect() : undefined"
+          >
+            {{
+              integration.settings?.tags.length > 0
+                ? 'Update'
+                : 'Connect'
+            }}
+          </el-button>
+        </div>
+      </div>
+    </template>
+  </app-drawer>
+</template>
+<script>
+export default {
+  name: 'AppStackOverflowConnectDrawer'
+}
+</script>
+<script setup>
+import {
+  defineEmits,
+  defineProps,
+  computed,
+  ref,
+  watch
+} from 'vue'
+import { useThrottleFn } from '@vueuse/core'
+import { CrowdIntegrations } from '@/integrations/integrations-config'
+import { useStore } from 'vuex'
+import Nango from '@nangohq/frontend'
+import AuthCurrentTenant from '@/modules/auth/auth-current-tenant'
+import config from '@/config'
+import isEqual from 'lodash/isEqual'
+import { IntegrationService } from '@/modules/integration/integration-service'
+
+const MAX_STACK_OVERFLOW_QUESTIONS = 10000
+
+const store = useStore()
+
+const tenantId = computed(() => AuthCurrentTenant.get())
+
+const props = defineProps({
+  modelValue: {
+    type: Boolean,
+    default: false
+  },
+  integration: {
+    type: Object,
+    default: () => {}
+  }
+})
+
+const emit = defineEmits(['update:modelValue'])
+const tags = computed(
+  () =>
+    props.integration.settings?.tags.map((i) => {
+      return {
+        value: i,
+        validating: false,
+        touched: true,
+        valid: true,
+        volumeTooLarge: false
+      }
+    }) || [{ value: '', loading: false }]
+)
+
+const keywords = computed(
+  () => props.integration.settings?.keywords || []
+)
+const calculateVolume = async () => {
+  const data = await IntegrationService.stackOverflowVolume(
+    modelKeywords.value.join(';')
+  )
+  return data
+}
+
+const model = ref(JSON.parse(JSON.stringify(tags.value)))
+const modelKeywords = ref(
+  JSON.parse(JSON.stringify(keywords.value))
+)
+const keywordsCount = computed(
+  () => modelKeywords.value.length
+)
+
+const isVolumeUpdating = ref(false)
+const isKeywordsValid = ref(true)
+
+const logoUrl =
+  CrowdIntegrations.getConfig('stackoverflow').image
+
+const hasFormChanged = computed(
```

Comment:
We now have a new approach for forms and forms validation. We are now using `vuelidate`, maybe it would help with these validations. You can check it in `eagle-eye-settings-drawer.vue` for example

However, this might be a major change. If you think it makes more sense, maybe we can create a task to update it later

Reply:
No problem! We can change it later